### PR TITLE
feat: optimise ux for slack component

### PR DESCRIPTION
**Pull request by chuang8511**
Because

- Now, there are several weird parts in Slack component.

This commit

- add user name in read task output, take out is_public_channel for TASK, and change line when posting message.

### CODE REVIEW

**Comment by donch1989** [comment]
@chuang8511 
Can you describe the weird parts in the commit message?

File: application/slack/v0/api_functions.go [hunk 1]
```diff
@@ -8,9 +8,9 @@ import (
 	"github.com/slack-go/slack"
 )
 
-func loopChannelListAPI(e *execution, isPublic bool, channelName string) (string, error) {
+func loopChannelListAPI(e *execution, channelName string) (string, error) {
 	var apiParams slack.GetConversationsParameters
-	setChannelType(&apiParams, isPublic)
+	apiParams.Types = append(apiParams.Types, "private_channel", "public_channel")
```
Comment (donch1989) [inline application/slack/v0/api_functions.go:13]: If the `types` is always "private_channel" and "public_channel," we can just make it be constant when sending the request to the Slack API in `GetConversationsForUserContext()`.

File: application/slack/v0/task_functions.go [hunk 2]
```diff
@@ -81,6 +117,40 @@ func (e *execution) readMessage(in *structpb.Struct) (*structpb.Struct, error) {
...
+
+	userIDs = removeDuplicateUserIDs(userIDs)
+	users, err := e.client.GetUsersInfo(userIDs...)
+
+	if err != nil {
+		return nil, err
+	}
+
+	for i, conversation := range readTaskResp.Conversations {
+		for _, user := range *users {
```
Comment (donch1989) [inline application/slack/v0/task_functions.go:139]: We can create a `map[string]string` to convert id to name, which will speed things up.

**Review by donch1989 — CHANGES_REQUESTED** [review]

File: application/slack/v0/api_functions.go (hunk 1 above)
Comment (chuang8511) [reply application/slack/v0/api_functions.go:13]: @donch1989 
Do you mean like these?
1. fork Slack package
2. modify the Slack package
3. Change the import source

Comment (chuang8511) [reply application/slack/v0/api_functions.go:13]: Except this, others are done. So, I request the review again!

Comment (donch1989) [reply application/slack/v0/api_functions.go:13]: Oh.. sorry, I wrote the wrong function name.
I mean you can have a global variable and use it like this

```
var types []string{"private_channel", "public_channel"}

---

apiParams.Types = types
```

**Review by donch1989 — APPROVED** [review]

**Pull request merged**